webhook.go

```diff
@@ -28,22 +30,93 @@ type DeliveryConfig struct {
 	AlternativeURLs []string `json:"alt_urls,omitempty"`
 }
 
+// WebhookConfig is a Webhook substructure with data related to event delivery.
+type WebhookConfig struct {
+	// ID is the configured webhook's name used to map hashed events to.
+	// Refer to the Hash substructure configuration for more details.
+	ID string `json:"id"`
+
+	// Accept is content type of outgoing events. The following content types are supported, otherwise
+	// a 406 response code is returned: application/octet-stream, application/jsonl, application/msgpack.
+	Accept string `json:"accept"`
+
+	// Secret is the string value.
+	// (Optional, set to "" to disable behavior).
+	Secret string `json:"secret,omitempty"`
+
+	// SecretHash is the hash algorithm to be used. Only sha256 HMAC and sha512 HMAC are supported.
+	// (Optional).
+	// The Default value is the sha512 HMAC.
+	SecretHash string `json:"secret_hash"`
+
+	// BatchHints is the substructure for configuration related to event batching.
+	// (Optional, if omited then batches of singal events will be sent)
+	// Default value will disable batch. All zeros will also disable batch.
+	BatchHints struct {
+		// MaxLingerDuration is the maximum delay for batching if MaxMesasges has not been reached.
+		// Default value will set no maximum value.
+		MaxLingerDuration time.Duration `json:"max_linger_duration"`
+		// MaxMesasges is the maximum number of events that will be sent in a single batch.
+		// Default value will set no maximum value.
+		MaxMesasges int `json:"max_messages"`
+	} `json:"batch_hints"`
+
+	// DNSSrvRecord is the substructure for configuration related to load balancing.
+	DNSSrvRecord struct {
+		// FQDNs is a list of FQDNs pointing to dns srv records
+		FQDNs []string `json:"fqdns"`
+		// LoadBalancingScheme is the scheme to use for load balancing. Either the
+		// srv record attribute `weight` or `priortiy` can be used.
+		LoadBalancingScheme string `json:"load_balancing_scheme"`
+	} `json:"dns_srv_record"`
+}
+
 // MetadataMatcherConfig is Webhook substructure with config to match event metadata.
 type MetadataMatcherConfig struct {
 	// DeviceID is the list of regular expressions to match device id type against.
 	DeviceID []string `json:"device_id"`
+
+	// Account is the list of regular expressions to match account type against.
+	Account []string `json:"metadata:/account"`
+
+	// Model is the list of regular expressions to match model type against.
+	Model []string `json:"metadata:/hw-model"`
+
+	// FirmwareName is the list of regular expressions to match firmware type against.
+	FirmwareName []string `json:"metadata:/fw-name"`
 }
```

Comment:
clients will be able to regex against the entire wrp message (scheme will be provided)